### DISCUSSION TITLE
Initial beta release!!!

### DISCUSSION THREAD

**Discussion by shuji-koike** [shuji-koike/csgo#31] [Announcements]
https://shuji-koike.github.io/dem/

https://github.com/shuji-koike/dem/assets/1072158/e5b9e6cc-51e7-4891-989d-1f84767d0fef

**Comment by billfreeman44** [comment]
Nice work! Just some comments at a glance

- Would recommend to make the "dots" for future state only
- Maybe be able to toggle off the dots instead? 
- Recommend removing leading 0 from timestamps like `01:55` as rounds aren't ever 10 min
- Play button / animations? 
- Render smokes as translucent grey and 450(?) in game unit radius

**Reply by shuji-koike** [reply]
@billfreeman44 thx for your advice!

> Would recommend to make the "dots" for future state only

Sounds right but it will require a re-render for each frame. Not sure whether that's acceptable.

> Maybe be able to toggle off the dots instead?

This might be a better solution.

> Recommend removing leading 0 from timestamps like 01:55 as rounds aren't ever 10 min

This was an easy fix. Already deployed!

> Play button / animations?

You can hit the space key to toggle play/pause.
But the playback speed may be wrong. It needs some adjustment baseed on the tickrate.
It's not complete so it's kind of a hidden feture for now.

> Render smokes as translucent grey and 450(?) in game unit radius

Yeah, I neet to make them scale to the right size for each map.
This will need some work but I will take care of it with top priority.

----

I definitely have to check out your [project](https://pureskill.gg/) too! looks amazing!!!

**Reply by billfreeman44** [reply]
for the scale you can probably use https://github.com/pnxenopoulos/awpy/blob/main/awpy/data/map/map_data.json

**Reply by shuji-koike** [reply]
Yeah! This is exactly what I was looking for!!!
ARIGATO!!!

**Reply by shuji-koike** [reply]
> Render smokes as translucent grey and 450(?) in game unit radius

I have fixed the scale for smokes🙌

Although, I'm still having probles detecting smoke Activity.
In some cases smokes are renderd as active (popped) mid air (before landing).
In other cases smokes are NOT renderd active even after landing. 😕

https://github.com/shuji-koike/dem/blob/79ae1b8cd2f996eb3efcd66a7c85d09c5d3cdb2c/parse.go#L226-L230
I'm tracking active smokes with a map based on `events.SmokeStart`.
might be a race condition problem. 

Another (huge) issue is CS2 volumetric smokes.
For now, CS2 smokes are rendered as same as CSGO smokes.
Seems not impossible but a huge challenge to properly render volumetric smokes.
https://discord.com/channels/901824796302643281/901824796784996374/1211587058758324284

**Reply by shuji-koike** [reply]
> In other cases smokes are NOT renderd active even after landing.

this was a race condition problem. fixed by setting MsgQueueBufferSize to zero 847e5dc498a1b6a1ee00d77ef9b7d031826af0a8

> In some cases smokes are renderd as active (popped) mid air (before landing).

this problem still remains. 😕
